spring-cloud-gcp-autoconfigure/pom.xml

```diff
@@ -63,12 +63,18 @@
             <groupId>com.google.cloud.sql</groupId>
             <artifactId>mysql-socket-factory</artifactId>
             <optional>true</optional>
+            <exclusions>
+                <exclusion>
+                    <groupId>mysql</groupId>
+                    <artifactId>mysql-connector-java</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
 
         <dependency>
             <groupId>mysql</groupId>
             <artifactId>mysql-connector-java</artifactId>
-            <optional>true</optional>
+            <scope>compile</scope>
```

Comment:
Why `compile`? Seems for me it is going to be included transitively when we add this ` spring-cloud-gcp-autoconfigure` dependency: https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html

> compile
This is the default scope, used if none is specified. Compile dependencies are available in all classpaths of a project. Furthermore, those dependencies are propagated to dependent projects.

Definitely must be `optional`. That's how BIG autocofigure module works.